components/centraldashboard/public/components/main-page.js

```diff
@@ -80,47 +81,64 @@ export class MainPage extends PolymerElement {
                     },
                 ],
             },
-            hideToolbar: {type: Boolean, value: false},
             sidebarItemIndex: {type: Number, value: 0},
             iframeUrl: {type: String, value: ''},
             buildVersion: {type: String, value: '0.4.1'},
             dashVersion: {type: String, value: VERSION},
+            inIframe: {type: Boolean, value: false, readOnly: true},
             _devMode: {type: Boolean, value: DEVMODE},
         };
     }
 
+
     /**
-   * Array of strings describing multi-property observer methods and their
-   * dependant properties
-   */
+     * Array of strings describing multi-property observer methods and their
+     * dependant properties
+     */
     static get observers() {
         return [
             '_routePageChanged(routeData.page)',
         ];
     }
 
+    ready() {
```

Comment:
Just curious, why is this being added asynchronously? Also, ready() is a void method so I don't think you need to wrap the call to `super.ready()` in a promise.

Reply:
`window.fetch` is async, do you have a better way in mind? (it helps to load assets async to allow parallelized dom rendering). Right, it just helps kick off a promise. It could be in or out. It just saves us an unnecessary semi-colon. ¯\_(ツ)_/¯

Reply:
Also @prodonjs I am unable to write code like:

```js
async ready() {
    const svg = await fetch(/* asset */).then(r => r.text())
    /* other code */
}
```
_**tl;dr**_ It doesn't like async / awaits

Reply:
What I meant was that the ready() method returns void and not a promise, so it executes synchronously and doesn't wait for any async promises to resolve. You could do the same thing without needing the outer Promise.resolve().

However, I don't think any of this is needed. We already load the kf-logo_64px.svg by virtual of our entry point `index.js`. I tested adding the same markup to the `#Narrow-Slider` as we have in the `#Logo`. It does what you would expect and only fetches the image once. Even if not, it's likely that a user's browser would cache this anyways and it's not a particularly large asset.

Reply:
How did you color it blue from the `image`? That should not be possible

Reply:
You're correct there is no way to change the look of the SVG if it's sourced from an image tag.

However, it looks like the `ready` body can be simplified to since super.ready() is synchronous and we don't need to wait for it to resolve.
```
ready() {
        super.ready();
        fetch('assets/kf-logo_64px.svg')
            .then((r) => r.text())
            .then((svg) => {
                this.$['Narrow-Slider'].querySelector('.Logo').innerHTML += [
                    svg,
                    `<figcaption>Kubeflow</figcaption>`,
                ].join('');
            });
```